src/apps/filters/h3.c

```diff
@@ -348,6 +361,328 @@ SUBCOMMAND(getIcosahedronFaces,
     return E_SUCCESS;
 }
 
+/// Traversal subcommands
+
+SUBCOMMAND(
+    gridDisk,
+    "Returns a JSON array of a H3 cells within 'k' steps of the origin cell") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    int k = 0;
+    Arg kArg = {.names = {"-k"},
+                .required = true,
+                .scanFormat = "%d",
+                .valueName = "distance",
+                .value = &k,
+                .helpText = "Maximum grid distance for the output set"};
+    Arg *args[] = {&gridDiskArg, &helpArg, &cellArg, &kArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    int64_t len = 0;
+    H3Error err = H3_EXPORT(maxGridDiskSize)(k, &len);
+    if (err) {
+        return err;
+    }
+    H3Index *out = calloc(len, sizeof(H3Index));
```

Comment:
Do we need to check for failed memory allocation here?

Reply:
Technically yes, but the app is going to die if this happens no matter what if this happens. But I'd also assume anybody on a machine that cannot allocate 8 bytes is having a much harder time, which is why I left that error handling out?

Reply:
This could be much larger than 8 bytes with larger values of `k`. I think the program is better behaved if it explicitly tells the user that memory allocation failed rather than risk what happens with segfaulting, as in that case the user would have a much harder time tracking down what happened (they would need to attach a debugger to find out the problem is the `k` value.)

Reply:
Addressed. There's now a print to `stderr` and an `exit` for each if the pointer is `NULL`.